pytest_html/resources/main.js

```diff
@@ -2,6 +2,12 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+//created to increase testability
+var  DOCUMENT = document;
+
+function setDocument(elem) {
+  DOCUMENT = elem;
+}
```

Comment:
I would agree that this is probably not particularly good practice. There are some useful suggestions [here](http://developer.fortnox.se/blog/testing-the-dom-in-javascript/) that basically propose simplifying the JavaScript code to not depend on accessing the DOM directly or by using a QUnit fixture to mock the HTML content.